packages/astro/src/runtime/client/visible.ts

```diff
@@ -16,11 +16,13 @@ export default async function onVisible(astroId: string, _options: HydrateOption
     }
   };
 
-  const io = new IntersectionObserver(([entry]) => {
-    if (!entry.isIntersecting) return;
-    // As soon as we hydrate, disconnect this IntersectionObserver for every `astro-root`
-    io.disconnect();
-    cb();
+  const io = new IntersectionObserver((entries) => {
```

Comment:
Turns out if you only listen if the first element is visible, it listens!

Reply:
Is there any reason we were only listening for the first match before?

Reply:
Oh! I uh... kinda forgot that this could multiple entries could ever be returned. Makes sense!

Reply:
I mean, that works for most scenarios! This is such an edge case. You had to have 2 instances of a component on-page, and the _first_ had to be hidden. Funny our docs had this